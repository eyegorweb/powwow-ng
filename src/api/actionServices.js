import { query } from './utils';

export async function getActionServices() {
  const queryStr = `query {
    __type(name: "ActionServiceEnum") {
      enumValues {
        name
      }
    }
  }`;
  const response = await query(queryStr);
  return response.data.__type.enumValues.map(e => e.name);
}
