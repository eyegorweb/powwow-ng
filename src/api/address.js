import axios from 'axios';

const address = axios.create({
  baseURL: 'https://api-adresse.data.gouv.fr',
});

export function searchAddress(q) {
  // evite une erreur dans l'appel api
  if (!q) return Promise.resolve([]);

  return (
    address
      .get('/search', { params: { q } })
      // TODO faire une fonction qui mape vers une version qui ressemble Ã  ce qu'on a en BDD
      .then(res => res.data.features.map(f => f.properties))
  );
}
