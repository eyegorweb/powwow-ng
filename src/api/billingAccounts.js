import { query } from './utils';

// TODO : verifier si il est nÃ©cessaire de passer des objet de partenaires , pkpas iun tableau d'ids ?
export async function fetchBillingAccounts(q, partners, { page, limit, partnerTypes }) {
  let partnersIds,
    partnerGqlParam = '';

  if (partners && partners.length > 0) {
    partnersIds = partners.map(i => `"${i.id}"`).join(',');
    partnerGqlParam = `, partyId:{in: [${partnersIds}]}`;
  }

  let partnerTypesGqlFilter = '';
  if (partnerTypes && partnerTypes.length) {
    const ids = partnerTypes.map(p => `${p.id}`).join(',');
    partnerTypesGqlFilter = `, partyType: {in: [${ids}]}`;
  }

  const queryStr = `
  query{
    customerAccounts(filter:{code: {contains: "${q}"}${partnerGqlParam}${partnerTypesGqlFilter}}, pagination: {limit: ${limit}, page: ${page}}, sorting: {name: ASC}) {
      total,
      items {
        id
        name
        code
        party {
          id
        }
      }
    }
  }
  `;
  const response = await query(queryStr);
  return response.data.customerAccounts.items;
}

export async function fetchBillibAccountForPartnerId(partnerId) {
  return await fetchBillingAccounts('', [{ id: partnerId }], { page: 0, limit: 50 });
}
