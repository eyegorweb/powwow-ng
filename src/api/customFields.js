import { query } from './utils';
import get from 'lodash.get';

export async function fetchCustomFields(partnerId) {
  const queryStr = `
  query{
    party(id: ${partnerId}) {
      custom1FieldLabel
      custom1FieldType
    	custom1FieldMandatory
      custom1ValidatedValuesAsList
    	custom2FieldLabel
      custom2FieldType
    	custom2FieldMandatory
      custom2ValidatedValuesAsList
    	custom3FieldLabel
      custom3FieldType
    	custom3FieldMandatory
      custom3ValidatedValuesAsList
    	custom4FieldLabel
      custom4FieldType
    	custom4FieldMandatory
      custom4ValidatedValuesAsList
    	custom5FieldLabel
      custom5FieldType
    	custom5FieldMandatory
      custom5ValidatedValuesAsList
    	custom6FieldLabel
      custom6FieldType
    	custom6FieldMandatory
      custom6ValidatedValuesAsList
    }
  }
  `;
  const response = await query(queryStr);
  const customFields = [];

  for (let i = 1, max = 6; i <= max; i++) {
    addCustomField(response, i);
  }

  function addCustomField(response, index) {
    const label = get(response, `data.party.custom${index}FieldLabel`);
    if (!label) return;

    const data = {
      id: partnerId + 'customField' + index,
      codeInOrder: 'custom' + index,
      code: 'customField' + index,
      label,
      type: get(response, `data.party.custom${index}FieldType`),
      value: get(response, `data.party.custom${index}ValidatedValuesAsList`),
      mandatory: get(response, `data.party.custom${index}FieldMandatory`),
      enteredValue: undefined,
    };
    customFields.push(data);
  }

  return customFields;
}

export async function createCustomField({ partyId, label, type, values, mandatoryVal }) {
  const valuesStr = values.map(v => `"${v}"`).join(',');
  const queryStr = `
  mutation {
    createCustomField(customFieldInput: {
      partyId: ${partyId}
      customFieldLabel: "${label}"
      customFieldType: ${type}
      customFieldListValues: [${valuesStr}]
      customFieldMandatory: ${mandatoryVal}
    }) {
      id
    }
  }
  `;
  const response = await query(queryStr);
  return response.data.createCustomField;
}
