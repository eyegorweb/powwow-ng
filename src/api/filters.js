import { query } from './utils';
import { capitalize } from 'lodash-es';

export async function fetchPartners(filter, pagination) {
  const paginationInfo = pagination
    ? `, pagination: {page: ${pagination.page}, limit: ${pagination.limit}}`
    : '';
  const response = await query(`
  query {
    partners(q: '${filter}'${paginationInfo}) {
      id
      name
    }
  }`);
  return response.data.partners;
}

export async function fetchDeliveryCountries(locale) {
  // fr -> nameFr
  const key = 'name' + capitalize(locale);

  return query(`
  query {
    countries(sorting: {${key}: ASC}) {
      total
      countries {
        id
        code
        ${key}
      }
    }
  }`).then(res =>
    res.data.countries.countries.map(country => ({
      name: country[key],
      code: country.code,
      id: country.id,
    }))
  );
}
