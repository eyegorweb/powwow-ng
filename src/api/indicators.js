import { delay } from './utils';

// MOCK temporaire en attendant l'api du backend
export async function fetchGetSimIndicators() {
  await delay(200);

  return {
    ordersValidated: 4,
    ordersInProgress: 1,
    nonConfirmedOrders: 5,
    faillingOrders: 8,
    ordersToBeValidated: 2,
  };
}
