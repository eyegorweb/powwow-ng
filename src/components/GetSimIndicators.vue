<template>
  <ul class="list-group bg-white">
    <li class="list-group-item">{{ $t('indicators.ordersToValidate') }}
      <div class="float-right text-danger">{{ ordersValidated }}</div>
    </li>
    <li class="list-group-item">{{ $t('indicators.ordersInProgress') }}
      <div class="float-right text-success">{{ ordersInProgress }}</div>
    </li>
    <li class="list-group-item">{{ $t('indicators.nonConfirmedOrders') }}
      <div class="float-right text-warning">{{ nonConfirmedOrders }}</div>
    </li>
    <li class="list-group-item">{{ $t('indicators.faillingOrders') }}
      <div class="float-right text-warning">{{ faillingOrders }}</div>
    </li>
    <li class="list-group-item">{{ $t('indicators.ordersToBeValidated') }}
      <div class="float-right text-warning">{{ ordersToBeValidated }}</div>
    </li>
  </ul>
</template>

<script>
import { fetchGetSimIndicators } from '@/api/indicators';

export default {
  data() {
    return {
      ordersValidated: 0,
      ordersInProgress: 0,
      nonConfirmedOrders: 0,
      faillingOrders: 0,
      ordersToBeValidated: 0,
    };
  },

  async mounted() {
    const result = await fetchGetSimIndicators();
    this.ordersValidated = result.ordersValidated;
    this.ordersInProgress = result.ordersInProgress;
    this.nonConfirmedOrders = result.nonConfirmedOrders;
    this.faillingOrders = result.faillingOrders;
    this.ordersToBeValidated = result.ordersToBeValidated;
  },
};
</script>

<style lang="scss" scoped>
.list-group {
  padding: 1.25rem 1rem;

  .list-group-item {
    border: none;
    padding: 0.75rem 0;
    border-bottom: 2px solid $medium-gray;

    &:last-child {
      border: none;
    }
  }
}
</style>
