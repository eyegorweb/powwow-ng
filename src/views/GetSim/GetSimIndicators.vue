<template>
  <ul class="list-group bg-white">
    <li class="list-group-item">{{ $t('indicators.averageProcessingTime') }}
      <div class="float-right text-success">{{ indicators.averageProcessingTime }}</div>
    </li>
    <li class="list-group-item">{{ $t('indicators.orderToBeConfirmedByBO') }}
      <div class="float-right text-success">{{ indicators.orderToBeConfirmedByBO }}</div>
    </li>
    <li class="list-group-item">{{ $t('indicators.ordersFailed') }}
      <div class="float-right text-danger">{{ indicators.ordersFailed }}</div>
    </li>
    <li class="list-group-item">{{ $t('indicators.ordersInProgress') }}
      <div class="float-right text-warning">{{ indicators.ordersInProgress }}</div>
    </li>
    <li class="list-group-item">{{ $t('indicators.ordersNotConfirmed') }}
      <div class="float-right text-warning">{{ indicators.ordersNotConfirmed }}</div>
    </li>
    <li class="list-group-item">{{ $t('indicators.ordersToBeConfirmed') }}
      <div class="float-right text-warning">{{ indicators.ordersToBeConfirmed }}</div>
    </li>
  </ul>
</template>

<script>
import { fetchGetSimIndicators } from '@/api/indicators';

export default {
  data() {
    return {
      indicators: {},
    };
  },

  async mounted() {
    this.indicators = await fetchGetSimIndicators();
  },
};
</script>

<style lang="scss" scoped>
.list-group {
  padding: 1.25rem 1rem;

  .list-group-item {
    border: none;
    padding: 0.75rem 0;
    border-bottom: 2px solid $medium-gray;

    &:last-child {
      border: none;
    }
  }
}
</style>
