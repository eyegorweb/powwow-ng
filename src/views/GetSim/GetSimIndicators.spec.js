import { mount } from '@vue/test-utils';
import GetSimIndicators from './GetSimIndicators';
import * as api from '@/api/indicators';

import { $t } from '@/../tests-utils';

const mocks = { $t };

describe('GetSimIndicators', () => {
  it('fetches indicators', async () => {
    const indicators = {
      ordersValidated: 5,
      ordersInProgress: 4,
      nonConfirmedOrders: 3,
      faillingOrders: 2,
      ordersToBeValidated: 1,
    };

    api.fetchGetSimIndicators = jest.fn();
    api.fetchGetSimIndicators.mockResolvedValue(indicators);

    const wrapper = mount(GetSimIndicators, { mocks });

    await wrapper.vm.$nextTick();
    await wrapper.vm.$nextTick();

    expect(api.fetchGetSimIndicators).toHaveBeenCalled();

    const expectedSnapshot = `<ul class="list-group bg-white"><li class="list-group-item">indicators.ordersToValidate
    <div class="float-right text-danger">5</div></li> <li class="list-group-item">indicators.ordersInProgress
    <div class="float-right text-success">4</div></li> <li class="list-group-item">indicators.nonConfirmedOrders
    <div class="float-right text-warning">3</div></li> <li class="list-group-item">indicators.faillingOrders
    <div class="float-right text-warning">2</div></li> <li class="list-group-item">indicators.ordersToBeValidated
    <div class="float-right text-warning">1</div></li></ul>`;

    expect(wrapper.html()).toEqual(expectedSnapshot);
  });
});
