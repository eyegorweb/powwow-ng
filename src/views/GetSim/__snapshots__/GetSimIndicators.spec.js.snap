// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetSimIndicators fetches indicators 1`] = `
<ul class="list-group bg-white">
  <li class="list-group-item">
    indicators.ordersToBeConfirmed
    <div class="float-right"><button class="btn btn-link p-0 text-danger">
        2
      </button></div>
  </li>
  <li class="list-group-item">
    indicators.ordersInProgress
    <div class="float-right"><button class="btn btn-link p-0 text-success">
        2
      </button></div>
  </li>
  <li class="list-group-item">
    indicators.ordersNotConfirmed
    <div class="float-right"><button class="btn btn-link p-0 text-warning">
        0
      </button></div>
  </li>
  <li class="list-group-item">
    indicators.ordersFailed
    <div class="float-right"><button class="btn btn-link p-0 text-warning">
        0
      </button></div>
  </li>
  <li class="list-group-item">
    indicators.averageProcessingTime
    <div class="float-right text-success"><button class="btn btn-link p-0 text-warning">
        0
      </button></div>
  </li>
  <li class="list-group-item">
    indicators.orderToBeConfirmedByBO
    <div class="float-right"><button class="btn btn-link p-0 text-warning">
        0
      </button></div>
  </li>
</ul>
`;
